identity/aziot-identityd/src/main.rs

```diff
@@ -57,21 +67,24 @@ async fn run() -> Result<(), Box<dyn std::error::Error>> {
 				device_id: device.device_id
 			};
 
+			// Only consider the previous Hub modules if the current and previous Hub devices match.
+			let mut use_prev = false;
```

Comment:
The prev_hub_mset can be set directly where this flag is set. We don't need this right?

Reply:
Both prev_local_mmap and prev_hub_mset are set from the call to convert_to_map on line 51. prev_local_mmap is needed regardless of provisioning status, while prev_hub_mset is only needed for provisioned devices. So prev_hub_mset is generated regardless of whether it's needed because device provisioning status isn't known until after convert_to_map is called.

IMO, it's more efficient (the list of principals only needs to iterate once) to generate them both from a single convert_to_map and ignore prev_hub_mset using this flag if it's not needed.